### PR TITLE
Remove media type options for pull operation

### LINKED ISSUE #130: `oras pull` shouldn't require `--media-type` or `--allow-all`
When using oras for local testing and validation, we shouldn't have to specify the `media-type`. Registries only support a single type for a given content addressable url. 
Requiring the -media-type only complicates the default learning experience

### PR DESCRIPTION
**Pull request by qweeah**
This PR removes two media type options and artifact will be pull with the stored media type.

Resolves #130

### CODE REVIEW

File: cmd/oras/pull.go [hunk 1]
```diff
@@ -96,20 +92,14 @@ func runPull(opts pullOptions) error {
...
-
 	resolver := newResolver(opts.username, opts.password, opts.insecure, opts.plainHTTP, opts.configs...)
 	store := content.NewFileStore(opts.output)
 	defer store.Close()
 	store.DisableOverwrite = opts.keepOldFiles
 	store.AllowPathTraversalOnWrite = opts.pathTraversal
 
 	pullOpts := []oras.PullOpt{
-		oras.WithAllowedMediaTypes(opts.allowedMediaTypes),
+		oras.WithAllowedMediaTypes([]string{}),
```
[the remark this answers is not in the document]
Comment (shizhMSFT) [reply cmd/oras/pull.go:102]: We can simply remove this option.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/oras/pull.go [hunk 2]
```diff
@@ -18,15 +18,13 @@ import (
...
-	manifestConfigRef  string
-	verbose            bool
-	cacheRoot          string
+	targetRef         string
+	keepOldFiles      bool
+	pathTraversal     bool
+	output            string
+	manifestConfigRef string
+	verbose           bool
+	cacheRoot         string
```
Comment (qweeah) [inline cmd/oras/pull.go:27]: Test 1

File: cmd/oras/pull.go [hunk 3]
```diff
@@ -18,15 +18,13 @@ import (
...
 
 	debug     bool
 	configs   []string
```
Comment (qweeah) [inline cmd/oras/pull.go:30]: test 2

**Pull request merged**